process_sample.smk

```diff
@@ -1,11 +1,24 @@
 import re
 from pathlib import Path
+import pysam
 
 
 configfile: "workflow/reference.yaml"         # reference information
 configfile: "workflow/config.yaml"            # general configuration
 shell.prefix(f"set -o pipefail; umask 002; ")  # set g+w
 
+def check_for_ml_tag(bams):
+    """
+    Check if any BAM in list of BAMs contains the Ml tag.
```

Comment:
Maybe it might be better to check that _every_ bam has the tag. If only 1/3 of the reads are tagged, it's not super useful.

Although, I'm not sure who we'd be protecting here by checking that every bam is tagged.

Reply:
I guess it depends on how the model is written. If reads are treated independently and untagged reads don't affect methylation probabilities on tagged reads then there's no reason to _not_ run the cpg/5mc step in the scenario where only one of three bams has the tag.